### ISSUE TITLE
Negative Monte Carlo pose loss during training

### ISSUE THREAD

**Issue by jinhong-ni** [tjiiv-cprg/EPro-PnP#40]
Dear authors,

Thanks for releasing your excellent work. 

I'm training EPro-PnP-6DoF following your instructions provided. I noticed that loss_mc sometimes becomes negative, resulting in the overall loss being negative as well. I believe this is caused by the logarithm in the computation for L_pred, as the log of the number between 0 and 1 is negative. Is this behavior expected? Could you please elaborate a little on the logic behind it?

Much appreciated your reply in advance!

**Comment by Lakonik** [comment]
Hi! This is the expected behavior. Although the loss is derived from KL divergence in the paper, it is equivalent to cross-entropy up to an additive constant. Since the (usually positive) constant is not included in the simplified loss, the loss could well be negative.

> <div align="center"><img style="background: white;" src="https://latex.codecogs.com/svg.latex?%5Cbegin%7Baligned%7D%0D%0AD_%5Ctext%7BKL%7D%20%26%3D%20%5Cint%20t(y)%20%5Cleft(%5Clog%7Bt(y)%7D%20-%20%5Clog%7Bp(y%7CX)%7D%20%5Cright)%20%5Cmathrm%7Bd%7D%20y%20%5C%5C%0D%0A%26%3D%20-%5Cint%20t(y)%20%5Clog%7Bp(y%7CX)%7D%20%5Cmathrm%7Bd%7D%20y%20%2B%20const%20%5C%5C%0D%0A%26%3D%20-%5Cint%20t(y)%20%5Clog%20%5Cfrac%7Bp(X%7Cy)%7D%7B%5Cint%20p(X%7Cy)%20%5Cmathrm%7Bd%7D%20y%7D%20%5Cmathrm%7Bd%7D%20y%20%2B%20const%20%5C%5C%0D%0A%26%3D%20-%5Cint%20t(y)%20%5Clog%20p(X%7Cy)%5Cmathrm%7Bd%7D%20y%20%2B%20%5Cleft(%5Clog%20%5Cint%20p(X%7Cy)%20%5Cmathrm%7Bd%7D%20y%5Cright)%20%5Cint%20t(y)%20%5Cmathrm%7Bd%7D%20y%20%2B%20const%20%5C%5C%0D%0A%26%3D%20-%5Cint%20t(y)%20%5Clog%20p(X%7Cy)%5Cmathrm%7Bd%7D%20y%20%2B%20%5Clog%20%5Cint%20p(X%7Cy)%20%5Cmathrm%7Bd%7D%20y%20%2B%20const%20%0D%0A%5Cend%7Baligned%7D%0D"></div>
> 
> _Originally posted by @Lakonik in https://github.com/tjiiv-cprg/EPro-PnP/issues/36#issuecomment-1236793129_

**Comment by jinhong-ni** [comment]
Thanks for the quick and informative response!
